### DISCUSSION TITLE
Is it possible to shorten this code?

### DISCUSSION THREAD

**Discussion by ghost** [pascalabcnet/pascalabcnet#2940] [Q&A] [answered]
```
{$apptype windows}

uses 
  Eto.Forms,
  Eto.Drawing,
  Eto;
  
type
  MyForm = class(Form)
  public
    constructor;
    begin
      Title := 'Table Layout';
      ClientSize := new Eto.Drawing.Size(600, 400);
      
      var tblLayout := new &TableLayout;
      tblLayout.Spacing := new Eto.Drawing.Size(5, 5);
      tblLayout.Padding := new Eto.Drawing.Padding(10, 10, 10, 10);
      var tblRow1 := new &TableRow;
      var lbl1 := new &Label;
      lbl1.Text := 'First Column';
      tblRow1.Cells.Add(new TableCell(lbl1, true));
      var lbl2 := new &Label;
      lbl2.Text := 'Second Column';
      tblRow1.Cells.Add(new TableCell(lbl2, true));
      var lbl3 := new &Label;
      lbl3.Text := 'Third Column';
      tblRow1.Cells.Add(new TableCell(lbl3, true));
      var tblRow2 := new &TableRow;
      var tbox1 := new &TextBox;
      tbox1.Text := 'Some text';
      tblRow2.Cells.Add(new TableCell(tbox1, true));
      var dropDown1 := new &DropDown;
      dropDown1.Items.Add('Item 1');
      dropDown1.Items.Add('Item 2');
      dropDown1.Items.Add('Item 3');
      tblRow2.Cells.Add(new TableCell(dropDown1, true));
      var chkbox1 := new &CheckBox;
      chkbox1.Text := 'A checkbox';
      tblRow2.Cells.Add(new TableCell(chkbox1, true));
      var tblRow3 := new &TableRow;
      tblRow3.ScaleHeight := true;
      tblLayout.Rows.Add(tblRow1);
      tblLayout.Rows.Add(tblRow2);
      tblLayout.Rows.Add(tblRow3);
      Content := tblLayout;
      
      Menu := new Eto.Forms.MenuBar;
      Menu.QuitItem := new Command((sender, e) -> Application.Instance.Quit());
      Menu.QuitItem.Text := 'Quit';
      Menu.QuitItem.Shortcut := Application.Instance.CommonModifier or Keys.Q;
    end;
  end;

begin
  (new Application).Run(new MyForm);
end.
```

Compared to the C# version, the Pascal version is too verbose. Sometimes, some syntactic sugar is useful.

https://github.com/picoe/Eto/blob/develop/samples/Tutorials/CSharp/Tutorial3/Main.cs

The `with` statement is interesting. But every time I tried to use it, I encountered compilation error:

`Cannot use 'with' and lambdas in the same function`

**Comment by SunSerega** [comment]
`with` mostly exists for compatibility with older Pascal dialects. What are you trying to use it for in this case? I don't think it would help you shorten the code in this case.

**Reply by ghost** [reply]
`with` is a perfect syntactic sugar for this. See:

```
with tblLayout do
begin
  Spacing := new Eto.Drawing.Size(5, 5);
  Padding := new Eto.Drawing.Padding(10, 10, 10, 10);
end;
```

**Reply by SunSerega** [reply]
Yeah, so you are turning 2 lines into 5 to not write `tblLayout.` before each property.

**Reply by SunSerega** [reply]
It's good when the expression in `with` is extremely long. But in most cases like that - you can just store that expression in a variable.

**Reply by ghost** [reply]
> you can just store that expression in a variable.

Don't understand. Please give example.

**Reply by SunSerega** [reply]
Likewise, I don't understand what you don't understand. I've at least tried to describe it, but in your case - I'm left only guessing.

**Reply by ghost** [reply]
> Likewise, I don't understand what you don't understand. I've at least tried to describe it, but in your case - I'm left only guessing.

Imagine with the code on https://github.com/pascalabcnet/pascalabcnet/discussions/2943 but you have to write `dialog.` that many times. But let's discuss about that on https://github.com/pascalabcnet/pascalabcnet/discussions/2945. Back to this issue, you have no solutions to shorten the code, don't you?

**Reply by SunSerega** [reply, marked as answer]
I don't think it can be made much shorter. But it can be far more readable if you add empty lines and reorder the statements to give them all a common structure.
